Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,196 @@
+It offers support for commonly used SSO protocols and is "reasonably" fast
+in adopting to protocol standard changes and extensions. This has been
+observed in the case of logout support (backend and frontend variants) in OIDC.
+
+It offers a concept of "Identity Brokering", where Keycloak is not just IdP
+but also "client" to other IdPs. This allows daisy chaining of identity
+federation. In this configuration it can work as a point of protocol
+transition between different supported SSO protocols (SAML, OAuth 2.0, etc.).
+
+Beyond this capability of using other IdPs as identity sources, it also supports
+using classic LDAP based IAM services as backend (OpenLDAP and Active Directory,
+e.g.).
+
+Keycloak's implementation makes some design decisions, that are specific
+to it and have consequences for clients of the service. E.g. Keycloak
+has a concept of management "Realms", which have their own specific
+set of HTTP API entrypoints, both for administration as well as for IdP
+requests.
+
+Commonly Keycloak realms can be used to map them 1:1 to user domains,
+but since Keycloak supports configuring multiple backend IdPs in a
+realm to be used for "Identity Brokering", there is always the
+possibility to create a kind of "proxy" realm to provide a single
+standard set of HTTP API endpoints for SSO clients (service providers)
+to avoid the need to frequently extend/reduce client service configuration
+whenever a new IdP federation needs to be added to Keycloak to onboard
+a new customer. This is relevant for services like OpenStack Keystone,
+which currently cannot be easily reconfigured for new SSO endpoints
+without restarting the service, making the service unavailable for
+a short span of time and increasing risk connected with service restarts.
+
+Since version 17, Keycloak claims that it's capability for
+"cloud native" deployments on Kubernetes has improved.
+
+[Keycloak is offering a REST API](https://www.keycloak.org/docs-api/20.0.1/rest-api/index.html)
+for all aspects of its administration interface.
+
+For storage of Keycloak configuration and local user metadata
+(e.g. from which external IdP a user account originally came from)
+Keycloak supports several SQL backends through JDBC. Thus
+it can be hooked up to a Postgres Database or to a
+MariaDB/Galera cluster e.g..
+
+### Zitadel
+
+Zitadel is a newer implementation of a SSO IdP. It is implemented
+in Go and under active development.
+
+It came to attention of the SCS project because it offers a
+fresh take of an organization focussed data model, which has
+the potential to simplify IdP federation to SCS customer domains
+in the following areas:
+* For client services (single set of HTTP API endpoints).
+* For SCS operators for provisioning customer [organizations](https://zitadel.com/docs/concepts/structure/organizations)
+  and robust configuraton by using templated client, role and mapping
+  configuration.
+* For SCS customers for a robust user experience for self servicing.
+
```

Comment:
```suggestion
* For client services (single set of HTTP API endpoints).
* For SCS operators for provisioning customer [organizations](https://zitadel.com/docs/concepts/structure/organizations)
  and robust configuraton by using templated client, role and mapping
  configuration.
* For SCS customers for a robust user experience for self servicing.

[Delegated Access Management](https://zitadel.com/docs/concepts/structure/organizations) reduces the management overhead compared to isolated realms.
Projects (Applications + Roles) can be maintained by one organization and delegated to be used by many other Organizations. 
Managers that receive granted Projects can assign users permissions to use the project.
```

Reply:
Thanks for the input, I've picked it up with minor adjustment in wording.